utils/githooks/hook_base.sh

```diff
@@ -31,6 +32,7 @@ export -f _git_diff_cached_files
 hook=${0##*/}
 rm -f ".${hook}"
 
+skip_list=()
```

Comment:
Unfortunately, this does not fix the problem in my testing: With or without this line, I get the same "unbound variable" error. Some reading shows that the `read -a v` builtin command unsets `v` at the beginning. If we really want to keep `set -u`, then the following works for me:

```
@@ -41,7 +41,7 @@ run-parts() {
         # don't run vim .swp files
         [ "${i%.sw?}" != "${i}" ] && continue
         skip_item=false
-        for skip in "${skip_list[@]}"; do
+        for skip in "${skip_list[@]:-}"; do
             if [[ "${i}" =~ ${skip} ]]; then
                 skip_item=true
                 echo "Skipping ${i}"
```

Reply:
Thanks. Pushed :)